### PR TITLE
Identifier validation

### LINKED ISSUE #243: Handling of invalid column names (sql keywords)
It would be a nice feature, if the compiler does either of following if column names or similar are named with invalid names in the sense of that those names will result in some sql exception, like for example, a column name like "group".

*Side note/Idea*

I actually thought, that some automatic prefixig could solve that as well. But would not be backwards compatible though...

### PR DESCRIPTION
**Pull request by sbosley**
Implements validation of table, view, and column names. If the name is a keyword that is not permitted, an error will be logged during code generation. Using any other keyword will log a warning.

Closes #243

### CODE REVIEW

File: squidb-processor/src/com/yahoo/squidb/processor/SqlUtils.java [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+    /**
+     * @return true if word is a SQLite keyword
+     */
+    public static boolean isKeyword(String word) {
+        return word == null || ALL_KEYWORDS.contains(word.toUpperCase());
+    }
+
+    /**
+     * @return true if word is a restricted SQLite keyword, i.e. cannot be used as a table or column name.
+     * If a word is a SQLite keyword, it *may* be used as a table or column name, but it is highly recommended that
```
Comment (jdkoren) [inline squidb-processor/src/com/yahoo/squidb/processor/SqlUtils.java]: Might want to copy or move this sentence to the javadoc for `isKeyword()`

**Comment by sbosley** [comment]
Tests are all green on my box.

File: squidb-processor/src/com/yahoo/squidb/processor/plugins/defaults/properties/generators/BasicPropertyGenerator.java [hunk 2]
```diff
@@ -67,8 +67,6 @@ public BasicPropertyGenerator(ModelSpec<?> modelSpec, VariableElement field, Apt
     private void validateColumnName() {
         if (columnName.indexOf('$') >= 0) {
```
Comment (jdkoren) [inline squidb-processor/src/com/yahoo/squidb/processor/plugins/defaults/properties/generators/BasicPropertyGenerator.java]: I wonder if we should move this to `checkIdentifier()` so it's enforced on tables and views, too.

Comment (sbosley) [inline squidb-processor/src/com/yahoo/squidb/processor/plugins/defaults/properties/generators/BasicPropertyGenerator.java]: Yeah that's probably a good idea.

**Comment by jdkoren** [comment]
LGTM

**Pull request merged**